mart/models/modular.py

```diff
@@ -232,11 +236,15 @@ def log_metrics(self, metrics, prefix="", prog_bar=False):
         def enumerate_metric(metric, name):
             # Metrics can have arbitrary depth.
             if isinstance(metric, torch.Tensor):
-                metrics_dict[name] = metric
+                # Ignore non-scalar results generated by Metrics, such as list of classes from MAP.
+                if metric.shape == torch.Size([]):
+                    metrics_dict[name] = metric
             else:
                 for k, v in metric.items():
                     enumerate_metric(v, f"{name}/{k}")
 
         enumerate_metric(metrics, prefix)
 
-        self.log_dict(metrics_dict, prog_bar=prog_bar)
+        # PyTorch Lightning: It is recommended to use `self.log('validation_metrics/acc', ..., sync_dist=True)`
+        # when logging on epoch level in distributed setting to accumulate the metric across devices.
+        self.log_dict(metrics_dict, prog_bar=prog_bar, sync_dist=True)
```

Comment:
I thought this wasn't necessary with torchmetrics?

Reply:
You are right! https://torchmetrics.readthedocs.io/en/stable/pages/lightning.html

Fixed.